tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,1637 @@
+                "id": self.id_index,
+                # For produce only mode just set them to the same value
+                "source_topic_name": name,
+                "target_topic_name": name,
+                "msgs_per_transaction": app_config.msg_per_txn,
+                "total_messages": msgs_per_topic,
+                "index": 0,
+                "processed_count": 0,
+                "partitions_eof": set(),
+                "last_message_ts": datetime.now().timestamp(),
+                "msgs_rate_ms": self.msgs_rate_ms,
+                "consume_timeout_s": app_config.consume_timeout_s,
+                "producer": None,
+                "consumer_config": {},
+                "reached_eof": False,
+                "terminate": False
+            }
+            t = TopicStatus(**topic_config)
+            t.producer = ck.Producer({
+                'bootstrap.servers': self.brokers,
+                'transactional.id': t.transaction_id
+            })
+            t.producer.init_transactions()
+            self.topics[t.target_topic_name] = t
+
+    #
+    # Produce functions
+    #
+    def produce(self, wait=True):
+        """Starts produce messages thread
+
+        Args:
+            wait (bool, optional): Wait for produce thread to finish or not.
+            Defaults to True.
+        """
+        self.logger.info("Start of sending messages")
+        self.produce_thread = self.create_thread(
+            self._send_transaction, thread_name="stream_produce_thread")
+        if wait:
+            self.produce_thread.join()
+        return
+
+    def _send_transaction(self, logger: logging.Logger,
+                          topic: TopicStatus) -> TopicStatus:
+        """
+            Transactional message sent. Uses simple indexed generator
+        """
+        def acked(err: ck.KafkaError, msg: ck.Message):
+            """
+                Unsafe callback that fills up delivery reports
+                Can be modified to send reports externally
+
+                err: error class from Kafka
+                msg: message that was sent
+            """
+            t = msg.topic()
+            pt = msg.partition()
+            pt = pt if pt else 'N'
+            k = msg.key().decode()
+            v = msg.value().decode()
+
+            if err is not None:
+                # log delivery error locally
+                logger.error(f"[{t}({pt}):{k}/{v}] {err.str()}")
+                # update delivery report
+                self.delivery_reports[f"{t}-{k}"] = {
+                    "latency": msg.latency(),
+                    "outcome": err.str()
+                }
+
+            else:
+                # Saving all delivery reports turned off for now
+                # self.delivery_reports[f"{t}-{k}"] = {
+                #     "partition": pt,
+                #     "latency": msg.latency(),
+                #     "offset": msg.offset(),
+                #     "outcome": "OK"
+                # }
```

Comment:
Any more explanation as to why is this disabled "for now"?

Reply:
In order to track on per-message while debugging, it is a good way to see each message timing. But for normal operation it will slow down the whole thing. It can be deleted, but I would leave it until real tests are created and tested

Reply:
I guess I'll enable it not to confuse the reader. Also, this function probably will be deleted at the end anyway since checking HWMs like this is very slow